ydb/core/mind/bscontroller/console_interaction.cpp

```diff
@@ -0,0 +1,209 @@
+            NTabletPipe::SendData(Self.SelfId(), *ConsolePipe, commitConfigEv.release());
+            return;
+        }
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerConsoleCommitResponse::TPtr &ev) {
+        STLOG(PRI_DEBUG, BS_CONTROLLER, BSC20, "Console commit config response", (Response, ev->Get()->Record));
+        auto& record = ev->Get()->Record;
+        auto status = record.GetStatus();
+        auto RetryConsoleCommit = std::make_unique<TEvBlobStorage::TEvControllerRetryConsoleCommit>();
+        switch(status) {
+            case NKikimrBlobStorage::TEvControllerConsoleCommitResponse::SessionMismatch:
+                MakeGetBlock();
+                MakeRetrySession();
+                return;
+            case NKikimrBlobStorage::TEvControllerConsoleCommitResponse::NotCommitted:
+                STLOG(PRI_ALERT, BS_CONTROLLER, BSC28, "Console commit config not commited");
+                return;
+            case NKikimrBlobStorage::TEvControllerConsoleCommitResponse::Committed:
+                if (GRPCSenderId) {
+                    auto response = std::make_unique<TEvBlobStorage::TEvControllerReplaceConfigResponse>();
+                    response->Record.SetStatus(NKikimrBlobStorage::TEvControllerReplaceConfigResponse::Success);
+                    Self.Send(*GRPCSenderId, response.release());
+                }
+                return;
+            default:
+                Y_FAIL("Unexpected console commit config response status: %d", status);
+        }
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerRetryConsoleCommit::TPtr &ev) {
+        STLOG(PRI_DEBUG, BS_CONTROLLER, BSC21, "Console retry commit config", (Request, ev->Get()->Record));
+        if (ev->Get()->Record.GetConfigVersion() != Self.ConfigVersion) {
+            STLOG(PRI_ALERT, BS_CONTROLLER, BSC23, "Config version mismatch in retry commit config.");
+            return;
+        }
+        auto commitConfigEv = std::make_unique<TEvBlobStorage::TEvControllerConsoleCommitRequest>();
+        commitConfigEv->Record.SetYAML(Self.YamlConfig);
+        NTabletPipe::SendData(Self.SelfId(), *ConsolePipe, commitConfigEv.release());
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerReplaceConfigRequest::TPtr &ev) {
+        STLOG(PRI_DEBUG, BS_CONTROLLER, BSC24, "Console replace config request", (Request, ev->Get()->Record));
+        auto& record = ev->Get()->Record;
+        auto response = MakeHolder<TEvBlobStorage::TEvControllerReplaceConfigResponse>();
+        if (!record.GetOverwriteFlag()) {
+            if (GRPCSenderId) {
+                response->Record.SetStatus(NKikimrBlobStorage::TEvControllerReplaceConfigResponse::OngoingCommit);
+                Self.Send(ev->Sender, response.Release());
+                return;
+            }
+        }
+        GRPCSenderId = ev->Sender;
+        if (!record.GetSkipConsoleValidation()) {
```

Comment:
and what if SkipConsoleValidation is true? do nothing, just hang on the query? :)